### PR TITLE
:hammer: Update Regression support

### PR DESCRIPTION
**Pull request by alafage**

### CODE REVIEW

File: torch_uncertainty/baselines/regression/mlp.py [hunk 1]
```diff
@@ -60,5 +60,6 @@ def __init__(
             dist_family=dist_family,
             is_ensemble=version in ENSEMBLE_METHODS,
             format_batch_fn=format_batch_fn,
+            save_in_csv=True,
```
[the remark this answers is not in the document]
Comment (alafage) [reply torch_uncertainty/baselines/regression/mlp.py:63]: You're right. In any case, baselines are going to disappear at some point. The new configs I added do not even use them

File: torch_uncertainty/metrics/regression/quantile_calibration.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        for i, conf in enumerate(self.conf_intervals):
+            b_min = iid_dist.icdf((1 - conf) / 2)
+            bound_log_prob = iid_dist.log_prob(b_min)
+            target_log_prob = dist.log_prob(target)
+            if reduce_event_dims:
+                bound_log_prob = bound_log_prob.sum(
+                    dim=list(range(-dist.reinterpreted_batch_ndims, 0))
+                )
+
+            corrects[..., i] = (bound_log_prob <= target_log_prob).float()
```
[the remark this answers is not in the document]
Comment (alafage) [reply torch_uncertainty/metrics/regression/quantile_calibration.py:62]: not really ids, it is just a mask when the target is within the bounds. Maybe `is_correct` instead?

File: torch_uncertainty/models/segmentation/unet/mimo.py [hunk 3]
```diff
@@ -26,8 +26,8 @@ def __init__(
     def forward(self, x: Tensor) -> Tensor:
         if not self.training:
             x = repeat(x, "b ... -> (m b) ...", m=self.num_estimators)
-        out = rearrange(x, "(m b) c ... -> b (m c) ...", m=self.num_estimators)
-        out = super().forward(out)
+        x = rearrange(x, "(m b) c ... -> b (m c) ...", m=self.num_estimators)
+        out = super().forward(x)
```
[the remark this answers is not in the document]
Comment (alafage) [reply torch_uncertainty/models/segmentation/unet/mimo.py:30]: I agree
